src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -1522,11 +1528,17 @@ JvmtiEnvBase::get_object_monitor_usage(JavaThread* calling_thread, jobject objec
     for (ObjectWaiter* waiter = mon->first_waiter();
          waiter != nullptr && (nWait == 0 || waiter != mon->first_waiter());
          waiter = mon->next_waiter(waiter)) {
+      JavaThread *w = mon->thread_of_waiter(waiter);
+      oop thread_oop = get_vthread_or_thread_oop(w);
+      bool is_virtual = java_lang_VirtualThread::is_instance(thread_oop);
+      if (is_virtual) {
+        skipped++;
```

Comment:
Do we really need to maintain `skipped`. Isn't not adding to `nWait` the same as skipping?

Reply:
Good suggestion, thanks. Fixed now.